### PR TITLE
Integrate comments into frontend

### PR DESCRIPTION
**Pull request by plausibly**
Integrate comments to frontend.

- Posts accurately reflect (total comment) count on preview
- Comments show up below a post, with the ability to create, edit, and delete comments.
- '3 dots' only appears to the author of a comment giving them the option to edit or delete
- Pagination for when there are many comments on a post
- Create comments input capped between 10-250 characters, button is only enabled after 10 characters are entered.

Note: Polling/fetch issue @danielduong8  can you look into this

image
image
image

Editing:
image

### CODE REVIEW

**Comment by plausibly** [comment]
NOTE: FIX ANCHOR ID

**Review by andy9775 — APPROVED** [review]

**Comment by danielduong8** [comment]
the polling issue you are experiencing will be resolved with the fix/polling pr

File: client/src/components/PostComment.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+  const deleteComment = () => {
+    api
+      .deleteComment(props.data.id)
+      .then((res) => {
+        if (res.status === 204) {
+          setMsg('Comment has been deleted. ');
+          props.setHasInteracted(true);
+        } else {
+          setMsg('Failed to delete comment.');
```
Comment (danielduong8) [inline client/src/components/PostComment.tsx:42]: write the error message in `res` to console so we have an easier time debugging

File: client/src/components/PostComment.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        }
+      })
+      .catch((err) => {
+        setMsg('Failed to delete comment. ');
```
Comment (danielduong8) [inline client/src/components/PostComment.tsx:46]: same here as well but with `err`

**Review by danielduong8 — CHANGES_REQUESTED** [review]
couple small fixes, but lgtm, merge afterwards

**Review by danielduong8 — APPROVED** [review]

**Pull request merged**